### PR TITLE
Added random-uniform-like operation on TensorMaps

### PR DESCRIPTION
**Pull request by abmazitov**
----
:books: Documentation preview :books:: https://equistore--196.org.readthedocs.build/en/196/

### CODE REVIEW

**Comment by Luthaf** [comment]
The linter is failing! You can run it locally with `tox -e lint`, and reformat the code automatically with `tox -e format`

File: python/tests/operations/random_like.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+import numpy as np
+
+import equistore
+
+
+DATA_ROOT = os.path.join(os.path.dirname(__file__), "..", "data")
+
+
+class TestRandomLike(unittest.TestCase):
```
Comment (DavideTisi) [inline python/tests/operations/random_like.py:12]: i know that it will be a pain but we are moving to pytest, and any new addition to the code have to have the test in pytest not unittest

File: python/src/equistore/operations/random_like.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+def random_uniform_like_block(
+    block: TensorBlock, parameters: Union[List[str], str] = None
+) -> TensorBlock:
+    """Return a new :py:class:`TensorBlock` with the same metadata as block, and all
+    values randomly sampled from the uniform distribution between 0 and 1.
+    :param block: Input block from which the metadata is taken.
+    :param parameters: Which gradient parameter to copy. If ``None`` (default)
+                       all gradient of ``tensor`` are present in the new tensor.
+                       If empty list ``[]`` no gradients information are copied.
+    :param requires_grad: If autograd should record operations for the returned tensor.
```
Comment (DavideTisi) [inline python/src/equistore/operations/random_like.py:36]: `required grad` is not a parameter of the function

File: python/tests/operations/random_like.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            self.assertTrue(np.all(rand_block.properties == block.properties))
+            self.assertEqual(len(rand_block.components), len(block.components))
+            self.assertTrue(
+                np.all(
+                    [
+                        np.all(rand_block.components[i] == block.components[i])
+                        for i in range(len(block.components))
+                    ]
+                )
+            )
```
Comment (DavideTisi) [inline python/tests/operations/random_like.py:34]: I would remove the check of the metadata and use before the loop
`python
assert equistore.equal(rand_tensor,tensor,only_metadata=True)
`
Like this it checks if the metadata are the same, then you can leave the loop to checks the values

Comment (DavideTisi) [reply python/tests/operations/random_like.py:34]: same thing for the gradinet, just leave the check of the values

File: python/tests/operations/random_like.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                            np.all(
+                                rand_gradient.components[i] == gradient.components[i]
+                            )
+                            for i in range(len(gradient.components))
+                        ]
+                    )
+                )
+
+                self.assertTrue(np.all(rand_gradient.data >= 0))
+                self.assertTrue(np.all(rand_gradient.data <= 1))
```
Comment (DavideTisi) [inline python/tests/operations/random_like.py:101]: same as before for the metadata

**Review by DavideTisi — CHANGES_REQUESTED** [review]
very good, i requested few changed in the test should be easy since you can reduce the number of check with `equistore.equal(only_metadata=True)`

File: python/tests/operations/random_like.py (hunk 3 above)
Comment (Luthaf) [reply python/tests/operations/random_like.py:34]: Is a gradinet a small gradient 😁 ? 🇮🇹 🤌

**Comment by abmazitov** [comment]
Ok, thank you for the response, I will do the changes

File: python/tests/operations/random_like.py (hunk 3 above)
Comment (DavideTisi) [reply python/tests/operations/random_like.py:34]: it is a gradient "gratinato"

**Comment by DavideTisi** [comment]
Some tests are failing because since `1a495992d6d6e82cc508fed4926a91babea311ac` 
`equal` does not have `only_metadata` anymore, but you have to use
`equistore.equal_metadata`

**Review by DavideTisi — CHANGES_REQUESTED** [review]
very good, two things one change the name of the file equal to the name of the function, just to be consistent to the rest of ops. And you should add the operation in the docs

**Comment by abmazitov** [comment]
> 

I was actually thinking to add another way of generating random array, namely with a gaussian distribution. Should make it in a separate file or put inside the random_like.py ?
